TELCODOCS-1744/TELCODOCS-1742:Fixed and known issues from the Telco Core team (#76998)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

==== Node

* Previously, when manually restarting the `kubelet` service on a node, some state files were deleted in anticipation of a node reboot. As the node is not rebooted in this scenario, the `kubelet` service re-assigns CPU resources for running workloads. As a result, the CPU configuration after the kubelet restart might differ from the initial `cpuset` configuration. With this update, the `cpuset` configuration is correctly restored after a kubelet restart. (link:https://issues.redhat.com/browse/OCPBUGS-24366[*OCPBUGS-24366*])

[discrete]
[id="ocp-4-16-node-tuning-operator-bug-fixes_{context}"]
==== Node Tuning Operator (NTO)

* Previously, the Performance Profile Creator (PPC) incorrectly populated the `metadata.ownerReferences.uid` field for Day 0 performance profile manifests. As a consequence, it was not possible to apply a performance profile at Day 0 without manual intervention. With this release, the PPC does not generate the `metadata.ownerReferences.uid` field for Day 0 manifests. As a result, you can apply a performance profile manifest at Day 0 as expected. (link:https://issues.redhat.com/browse/OCPBUGS-29751[*OCPBUGS-29751*])

> * Previously, the `tuned` and `irqbalanced` daemons modified the Interrupt Request (IRQ) CPU affinity configuration. As a consequence, conflicts in the IRQ CPU affinity configuration might cause unexpected behaviour after a {sno} node restart. With this release, only the `irqbalanced` daemon determines IRQ CPU affinity configuration.(link:https://issues.redhat.com/browse/OCPBUGS-26400[*OCPBUGS-26400*])

Author: @yanirq - can you please review this bug fix release note text for OCPBUGS-26400? Thanks!

Reviewer: LGTM

[discrete]
[id="ocp-4-16-node-tuning-operator-bug-fixes_{context}"]
==== Node Tuning Operator (NTO)

* Previously, the Performance Profile Creator (PPC) incorrectly populated the `metadata.ownerReferences.uid` field for Day 0 performance profile manifests. As a consequence, it was not possible to apply a performance profile at Day 0 without manual intervention. With this release, the PPC does not generate the `metadata.ownerReferences.uid` field for Day 0 manifests. As a result, you can apply a performance profile manifest at Day 0 as expected. (link:https://issues.redhat.com/browse/OCPBUGS-29751[*OCPBUGS-29751*])

* Previously, the `tuned` and `irqbalanced` daemons modified the Interrupt Request (IRQ) CPU affinity configuration. As a consequence, conflicts in the IRQ CPU affinity configuration might cause unexpected behaviour after a {sno} node restart. With this release, only the `irqbalanced` daemon determines IRQ CPU affinity configuration.(link:https://issues.redhat.com/browse/OCPBUGS-26400[*OCPBUGS-26400*])

> * Previously, when restarting a {sno} cluster with a performance profile applied, a race condition caused systemd processes to remain in the root directory instead of being moved to `/sys/fs/cgroup/cpuset/system.slice`. With this release, systemd processes are moved to `/sys/fs/cgroup/cpuset/system.slice` as expected following a {sno} node restart. (link:https://issues.redhat.com/browse/OCPBUGS-31692[*OCPBUGS-31692*])

Author: @rbaturov - can you please review this 4.16 bug fix release note? Thanks!

Reviewer 2: Looks good to me.

[id="ocp-4-16-scalability-and-performance-bug-fixes_{context}"]
==== Scalability and performance

* Previously, the NUMA Resources Operator did not correctly handle `KubeletConfig` objects featuring an empty payload. As a consequence, the NUMA resources controller reported a `CrashLoopBackOff` status when applying a `NUMAResourcesOperator` resource in this scenario. With this release, improved filtering logic ensures that the NUMA Resources Operator handles `KubeletConfig` objects with empty payloads as expected. (link:https://issues.redhat.com/browse/OCPBUGS-30236[*OCPBUGS-30236*])

> 

Author: @ffromani - can you review this bug fix text please?

Reviewer 3: LGTM

[id="ocp-telco-core-4-16-known-issues_{context}"]

* When a worker node's Topology Manager policy is changed, the NUMA-aware secondary pod scheduler does not respect this change, which can result in incorrect scheduling decisions and unexpected topology affinity errors. As a workaround, restart the NUMA-aware scheduler by deleting the NUMA-aware scheduler pod. (link:https://issues.redhat.com/browse/OCPBUGS-34583[*OCPBUGS-34583*])

> * Due to an issue with Kubernetes, the CPU Manager is unable to return CPU resources from the last pod admitted to a node to the pool of available CPU resources. These resources are allocatable if a subsequent pod is admitted to the node. However, this in turn becomes the last pod, and again, the CPU manager cannot return this pod's resources to the available pool.

Reviewer 4: Is there a bug to link to?

Reviewer 4: I find this one a bit confusing, and a touch convoluted. But, I have no idea how to reword it. 

```suggestion
* Due to an issue with Kubernetes, the CPU Manager is unable to return to the pool of available CPU resources the resources from the last pod admitted to a node. The resources from the last pod are reallocatable if a subsequent pod is admitted to the node. However, this pod becomes the last pod, and again, the CPU manager cannot return this pod's resources to the available pool.
```

Author: It is a confusing one. I've made minor changes based on your feedback, thanks!

+

> This issue affects CPU load balancing features because these features depend on the CPU Manager releasing CPUs to the available pool. Consequently, non-guaranteed pods might run with a reduced number of CPUs. As a workaround, schedule a pod with a `best-effort` CPU Manager policy on the affected node. This pod will be the last admitted pod and this ensures the resources will be correctly released to the available pool.(link:https://issues.redhat.com/browse/OCPBUGS-17792[*OCPBUGS-17792*])

Reviewer 4: ```suggestion
This issue affects CPU load balancing features, which depend on the CPU Manager releasing CPUs to the available pool. Consequently, non-guaranteed pods might run with a reduced number of CPUs. As a workaround, schedule a pod with a `best-effort` CPU Manager policy on the affected node. This pod will be the last admitted pod and this ensures the resources will be correctly released to the available pool.(link:https://issues.redhat.com/browse/OCPBUGS-17792[*OCPBUGS-17792*])
```

[id="ocp-telco-core-4-16-known-issues_{context}"]

> * When a worker node's Topology Manager policy is changed, the NUMA-aware secondary pod scheduler does not respect this change, which can result in incorrect scheduling decisions and unexpected topology affinity errors. As a workaround, restart the NUMA-aware scheduler by deleting the NUMA-aware scheduler pod.

Author: @ffromani - can you review this known issue text please?

Reviewer 3: LGTM

* Due to an issue with Kubernetes, the CPU Manager is unable to return CPU resources from the last pod admitted to a node to the pool of available CPU resources. These resources are allocatable if a subsequent pod is admitted to the node. However, this in turn becomes the last pod, and again, the CPU manager cannot return this pod's resources to the available pool.
+
This issue affects CPU load balancing features because these features depend on the CPU Manager releasing CPUs to the available pool. Consequently, non-guaranteed pods might run with a reduced number of CPUs. As a workaround, schedule a pod with a `best-effort` CPU Manager policy on the affected node. This pod will be the last admitted pod and this ensures the resources will be correctly released to the available pool.(link:https://issues.redhat.com/browse/OCPBUGS-17792[*OCPBUGS-17792*])

> * After applying a `SriovNetworkNodePolicy` resource, the CA certificate might be replaced during SR-IOV Network Operator webhook reconciliation. As a consequence, you might see `unknown authority` errors when applying SR-IOV Network node policies. As a workaround, try to re-apply the failed policies. (link:https://issues.redhat.com/browse/OCPBUGS-32139[*OCPBUGS-32139*])

Author: @cgoncalves - Can you please review this known issue release note -thanks

Reviewer 5: LGTM.
@zeeke WDYT?

Reviewer 6: LGTM

> * If you delete a `SriovNetworkNodePolicy` resource for a virtual function with a `vfio-pci` driver type, the SR-IOV Network Operator is unable to reconcile the policy. As a consequence the `sriov-device-plugin` pod enters a continuous restart loop. As a workaround, delete all remaining policies affecting the physical function, then re-create them. (link:https://issues.redhat.com/browse/OCPBUGS-34934[*OCPBUGS-34934*])

Author: @cgoncalves - can you please reveiw this known issue release note?

Reviewer 5: LGTM.
@zeeke WDYT?

Reviewer 6: Looks great
